### ISSUE TITLE
False positive: vertical_parameter_alignment with specific function name

### ISSUE THREAD

**Issue by MrCOBA** [realm/SwiftLint#5037]
### Describe the bug

Hi! Found a new false positive case of vertical_parameter_alignment. The problem is reproduced on SwiftLint version `0.52.2`.
I cannot explain this error, in this case it occurs :
```swift
    func сheckprofileexistence(_ param: Bool,
                               param1: Bool,
                               param2: Bool,
                               param3: Bool) -> Bool {
        return false
    }
```
In the following example, no:
```Swift
    func foooooooooooooooooooo(_ param: Bool,
                               param1: Bool,
                               param2: Bool,
                               param3: Bool) -> Bool {
        return false
    }
```
### Environment

SwiftLint version `0.52.2`
Installation method used swiftlint binary
Paste your configuration file:
```
only_rules:
  - vertical_parameter_alignment
```
Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
Which Xcode version are you using? 14.2
Do you have a sample that shows the issue?
```Swift
    func сheckprofileexistence(_ param: Bool,
                               param1: Bool,
                               param2: Bool,
                               param3: Bool) -> Bool {
        return false
    }
```

**Comment by SimplyDanny** [comment]
The first character of your function name is a Cyrillic `с` (U+0441). It occupies 4 byte more than an ASCII `c`.

Admittedly, SwiftLint shouldn't care for the number of bytes but for the length of visible characters only. So this is actually a bug we have also seen in other contexts.

I don't think you rely on this specific character, however. So replacing it with an ASCII `c` should silence the warning for you.

**Comment by SimplyDanny** [comment]
Does my previous comment resolve the issue, @MrCOBA?
